### ISSUE TITLE
Say Block For Stage

### ISSUE THREAD

**Issue by TheBrokenRail** [LLK/scratch-gui#944]
### Curent Behavior

_Say Bubble Appears In Top-Left corner of the screen._

### Requested Behavior

_An ask input appears, but instead of a text box it is just text with an ok button._

### Why

_It will make displaying messages easier._

**Comment by Grinthy** [comment]
Uhh, you're not making sense to me. So, correct me if I'm wrong, but lemme try to summarize:
- The say bubble should go in the top-left corner, right?
- I'm in confusion about the "ask [] and wait" and "ask [] and wait () sec" block behavior. Can you clarify?

Thanks,
-- Ryan

**Comment by Kenny2github** [comment]
@Grinthy no, what the user is saying is:
### What currently happens
a "say" bubble appears in the top left corner
### What is being requested
the box used for "ask" blocks appears, but without the text input, and with just the "ok" button on the right.

**Comment by redmangospros** [comment]
I don't think that this should be addded because a say nlock and a text in the background will look werid.
Thats why you can add text in a backdrop. @CosmicWebServices

**Comment by TheBrokenRail** [comment]
Heres a Concept Image
### Current
![current](https://user-images.githubusercontent.com/17478432/33236253-eaddf222-d21b-11e7-8af8-61c492d2f522.PNG)
### Proposed
![proposed](https://user-images.githubusercontent.com/17478432/33236255-f20be8ba-d21b-11e7-9fb8-a7c6281593c9.PNG)

**Comment by sjhuang26** [comment]
This seems like a good idea. It would make things easier for beginners.

But why need the check mark? If I understand correctly, this is a "say ----- for () seconds" block for the stage, and normal speech bubbles don't have check marks. The message could simply automatically disappear after the specified number of seconds.

**Comment by redmangospros** [comment]
Acutully, i chanve my mind. It looks good. :D

**Comment by mrjacobbloom** [comment]
This could also apply to hidden sprites

**Comment by Kenny2github** [comment]
@mrjacobbloom currently hidden sprites don't say anything though... if this were to apply to hidden sprites, then it would break some projects.

**Comment by dalearn** [comment]
Maybe we could have a "Say" block similar to the Scratch 2.0 "Ask and Wait" block that displays a generic message that is unrelated to any object.

**Comment by redmangospros** [comment]
@dalearn  good idea.

**Comment by dalearn** [comment]
@redmangospros but if we do that, should we put the "Ask and Wait" block back in to complement it?  Was there any discussion on its removal initially?

**Comment by redmangospros** [comment]
On removal I'm not sure. I think we could put the ask and wait back into component.

**Comment by dalearn** [comment]
It appears that I had an old version or something.  It's in there in the latest version

**Comment by kyleplo** [comment]
Well, this actually sound like the JavaScript alert, confirm and prompt functions. For alert, you could use @TheBrokenRail 's idea:
![alert](https://user-images.githubusercontent.com/17478432/33236255-f20be8ba-d21b-11e7-9fb8-a7c6281593c9.PNG)
For confirm, you could use this:
![confirm](https://user-images.githubusercontent.com/31634240/33292813-8b779256-d398-11e7-80f3-70571aaba760.PNG)
And prompt would be like the current "ask" block
Sorry, my image editing skills are bad.

**Comment by redmangospros** [comment]
@kyleplo I argge with your idea and change of @TheBrokenRail.
@dalearn Thanks for classifying that you have a old Version

**Comment by CosmicWebServices** [comment]
No longer confused... good idea... maybe an extension? btw kinda off topic will there be an extension API open to the public?

**Comment by redmangospros** [comment]
I think we need a scratch team member to answer yoir question of a extestion api to pybilc.
So i will call @thisandagain.


I cpuld deplovp a extestion with help though as ypu said cosmic. But i will need help. Though this could ve added in the scratch blocks without the need of a

**Comment by TheBrokenRail** [comment]
Here's a demo implementation of the feature I made:
https://github.com/TheBrokenRail/scratch-gui/tree/say-stage which depends on https://github.com/TheBrokenRail/scratch-vm/tree/say-stage

**Comment by thisandagain** [comment]
I like this idea @TheBrokenRail. I'll make sure we discuss it the next time we look at the feature backlog.

**Comment by redmangospros** [comment]
Great.

**Comment by CosmicWebServices** [comment]
@kyleplo what would a confirm block look like?

**Comment by kyleplo** [comment]
@CosmicWebServices 
![confirm](https://user-images.githubusercontent.com/31634240/33401753-67986798-d528-11e7-8fd4-97afa4695e20.png)

**Comment by redmangospros** [comment]
Cool, But Why Add The Ask Text () Block, when you can just use yes or no,
ask and wait or bnumber.

**Comment by Grinthy** [comment]
This thing exploded.

**Comment by redmangospros** [comment]
Lol

**Comment by apple502j** [comment]
@thisandagain This is not a suggestion, but a bug:
![](https://screenshotscdn.firefoxusercontent.com/images/f2512e4e-3af9-4173-a6d7-6711811c94bd.png)

(To repro:  just share the love.

**Comment by thisandagain** [comment]
@apple502j The "bug" part of this (deviation from 2.0) is being tracked here: https://github.com/LLK/scratch-render/issues/373

This was originally created as a feature request to add this functionality so I prefer we keep this issue focused on that.

**Comment by TheBrokenRail** [comment]
Any update on this?

**Comment by Samq64** [comment]
For the original suggestion, I think something like @TheBrokenRail's implementation would work well if it was a bit smaller and also worked for sprites, but it should probably be separate from the say block. Maybe something like this?

![display block](https://user-images.githubusercontent.com/81489795/146070057-60862f99-632e-4905-82fb-64edac34ab83.png)


For @kyleplo's suggestion, I like the idea of making the ask block a drop-down, but I'm not sure the second answer block. Maybe a better idea since the ask system will likely be changed quite a bit already if this gets implemented is to also implement [this Scratch forum suggestion](https://scratch.mit.edu/discuss/topic/313555/) which turns the ask block into a reporter and removes (or hides) the answer reporter, that way no new blocks are needed and one is removed without taking away any functionality. It would also be nice if you could ask for different types of data through the drop-down like in the image below.

![ask block reporters](https://user-images.githubusercontent.com/81489795/146069889-55015d0d-5cf1-4683-a3bc-6bc4d368d249.png)
